pkg/ebpf/tracee.go

```diff
@@ -831,28 +859,29 @@ func (t *Tracee) initBPF() error {
 		return err
 	}
 
-	if t.config.Capture.NetIfaces != nil || t.config.Debug {
-		for _, iface := range t.config.Capture.NetIfaces {
-			ingressHook, err := t.attachTcProg(iface, bpf.BPFTcIngress, "tc_ingress")
-			if err != nil {
-				return err
-			}
-
-			egressHook, err := t.attachTcProg(iface, bpf.BPFTcEgress, "tc_egress")
-			if err != nil {
-				return err
-			}
-
-			tcProbe := netProbe{
-				ingressHook: ingressHook,
-				egressHook:  egressHook,
-			}
-			t.tcProbe = append(t.tcProbe, tcProbe)
+	if t.eventsToTrace[NetPacket] && len(t.config.Filter.NetFilter.InterfacesToTrace) == 0 {
+		return fmt.Errorf("couldn't trace %s event, missing interface", EventsDefinitions[NetPacket].Name)
```

Comment:
This is the error message I get if I try to trace the `net_event` event without using the `-t net=<iface`.
This is not a clear message to get, because I don't get from it that I should add a tracing interface.
Try to catch the case of missing tracing interfaces before this point and giving a self clear error message for the user to understand what he should do to solve the problem.